### PR TITLE
Makman2/nextgen3-6

### PR DESCRIPTION
**Pull request by Makman2**

### CODE REVIEW

File: coalib/core/DependencyTracker.py [hunk 1]
```diff
@@ -97,6 +97,65 @@ def get_dependencies(self, dependant):
...
+            A set of dependants.
+        """
+        dependants = set()
+
+        def append_to_dependants(prev, nxt):
+            dependants.add(nxt)
+
+        traverse_graph(
+            [dependency],
+            lambda node: self._dependency_dict.get(node, frozenset()),
```
Comment (Nosferatul) [inline coalib/core/DependencyTracker.py]: is there any difference between frozenset and set except set has mutating methods added?

**Comment by Nosferatul** [comment]
ack feda9df

File: coalib/core/DependencyTracker.py (hunk 1 above)
Comment (Makman2) [inline coalib/core/DependencyTracker.py]: no that's the only difference :+1:
Actually I thought this gives a slight performance improvement as frozenset doesn't require an extra allocation, though the same would apply to a normal set, as it is reserved once before `traverse_graph` is called for the passed lambda. If you want I change it to set, it doesn't really matter^^

Comment (Makman2) [inline coalib/core/DependencyTracker.py]: Okay there still a little improvement: If somebody has used an empty frozenset before in the program, it's reused here. But this is such a super super minor mini improvement^^

Comment (Nosferatul) [inline coalib/core/DependencyTracker.py]: it's fine 👍

**Comment by Nosferatul** [comment]
ack 6adec40

**Comment by Makman2** [comment]
reack 8e3c280, 8dbd5e6

**Comment by Makman2** [comment]
@rultor merge

**Pull request merged**